plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -76,29 +94,45 @@ describe('KubernetesProxy', () => {
     return req;
   };
 
-  const clusterSupplier: jest.Mocked<KubernetesClustersSupplier> = {
-    getClusters: jest.fn(),
-  };
+  const setupProxyPromise = ({
+    proxyPath,
+    requestPath,
+    headers,
+  }: {
+    proxyPath: string;
+    requestPath: string;
+    headers?: Record<string, string>;
+  }) => {
+    const app = express().use(
+      Router()
+        .use(proxyPath, proxy.createRequestHandler({ permissionApi }))
+        .use(errorHandler()),
+    );
 
-  const permissionApi: jest.Mocked<PermissionEvaluator> = {
-    authorize: jest.fn(),
-    authorizeConditional: jest.fn(),
-  };
+    const requestPromise = request(app).get(proxyPath + requestPath);
 
-  const authTranslator: jest.Mocked<KubernetesAuthTranslator> = {
-    decorateClusterDetailsWithAuth: jest.fn(),
+    if (headers) {
+      for (const [headerName, headerValue] of Object.entries(headers)) {
+        requestPromise.set(headerName, headerValue);
+      }
+    }
+
+    // Let this request through so it reaches the express router above
+    worker.use(rest.all(requestPromise.url, (req: any) => req.passthrough()));
+
+    return requestPromise;
   };
 
   beforeEach(() => {
     jest.resetAllMocks();
     proxy = new KubernetesProxy({ logger, clusterSupplier, authTranslator });
+    permissionApi.authorize.mockReturnValue(
+      Promise.resolve([{ result: AuthorizeResult.ALLOW }]),
```

Comment:
While we're here, I think this version is a little tighter:
```suggestion
    permissionApi.authorize.mockResolvedValue(
      [{ result: AuthorizeResult.ALLOW }],
```